### ISSUE TITLE
New install doesn't allow admin user creation

### ISSUE THREAD

**Issue by ssilvert** [keycloak/keycloak#35675]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

I just downloaded the nightly build, started the server, and went to http://localhost:8180

Instead of the welcome page for adding the initial admin user, I got:
![image](https://github.com/user-attachments/assets/314f769b-c60b-4a5e-8024-f64cbfcbd950)

Clicking on the link didn't help.

Also, adding the environment variables didn't fix the problem.  Even with these set, I still can't log in.
```
C:\keycloak-nightly\keycloak-999.0.0-SNAPSHOT\bin>set |grep BOOTSTRAP
KC_BOOTSTRAP_ADMIN_PASSWORD=admin
KC_BOOTSTRAP_ADMIN_USERNAME=admin
```

This is on Windows, so it could be a Windows-only issue.

### Version

nightly

### Regression

- [X] The issue is a regression

### Expected behavior

Allow creation of initial admin user.

### Actual behavior

See above

### How to Reproduce?

See above

**Comment by shawkins** [comment]
Duplicate of https://github.com/keycloak/keycloak/issues/35526

**Comment by vmuzikar** [comment]
I don't think it's a duplicate.  It's not about the wording (which was addressed by #35526). It's about the welcome page no allows creating an admin user using a form. I notice the issue too.

**Comment by stianst** [comment]
Can confirm I can't create a admin on `http://localhost:8080` either. 

Environment variable approach works though for me, but you can't do the following on Linux at least, which is expected:

```
KC_BOOTSTRAP_ADMIN_PASSWORD=admin
KC_BOOTSTRAP_ADMIN_USERNAME=admin
kc.sh start-dev
```

You need to do:

```
export KC_BOOTSTRAP_ADMIN_PASSWORD=admin
export KC_BOOTSTRAP_ADMIN_USERNAME=admin
kc.sh start-dev
```

Or:

```
KC_BOOTSTRAP_ADMIN_PASSWORD=admin KC_BOOTSTRAP_ADMIN_USERNAME=admin kc.sh start-dev
```

**Comment by stianst** [comment]
Strange this isn't caught by the test: https://github.com/keycloak/keycloak/blob/main/tests/base/src/test/java/org/keycloak/test/welcomepage/WelcomePageTest.java

**Comment by stianst** [comment]
Looks like a regression caused by https://github.com/keycloak/keycloak/commit/0eb0281bf222d7781dbf2bdc084a4f73f06c8bc7

One the following line:
https://github.com/keycloak/keycloak/blob/8f2c3a744713db18033f1b4f2f004eeb70b1fb08/services/src/main/java/org/keycloak/services/resources/WelcomeResource.java#L275

On Chrome I get `localAddress=0:0:0:0:0:0:0:1` while on HtmlUnit and Firefox it's `127.0.0.1`. That explains why WelcomePageTest passes.

The value from Chrome (I guess anytime IPv6 is used) does not match the expected value in the following code:

https://github.com/keycloak/keycloak/blob/8f2c3a744713db18033f1b4f2f004eeb70b1fb08/services/src/main/java/org/keycloak/utils/SecureContextResolver.java#L81-L83

Which explains why it doesn't let the user set a password.

A workaround is to use `http://127.0.0.1` instead of `http://localhost` at least on Chrome.

**Comment by Eddyhanderson** [comment]
Hi @shawkins , hope you're well

I'm having the same issue even after the fix #35737  
After debugging i noted that variable  `address` didn't start with "[", instead had value `0:0:0:0:0:0:0:1`, reason why it didn't pass to regex check, hence giving the same error. 

![ipv6-issue](https://github.com/user-attachments/assets/088b1a0b-394e-4354-adf5-e7793baeb84c)

**Comment by jonkoops** [comment]
@Eddyhanderson if you are still experiencing issues please file a new issue with steps to reproduce. We are unable to keep track of comments on closed issues.
